src/main/java/io/confluent/kafkarest/KafkaRestApplication.java

```diff
@@ -62,70 +69,39 @@ public void setupResources(Configurable<?> config, KafkaRestConfig appConfig) {
    * Helper that does normal setup, but uses injected components so their configs or implementations
    * can be customized for testing. This only exists to support TestKafkaRestApplication
    */
-  protected void setupInjectedResources(
-      Configurable<?> config, KafkaRestConfig appConfig,
-      ZkUtils zkUtils, MetadataObserver mdObserver,
-      ProducerPool producerPool,
-      ConsumerManager consumerManager,
-      SimpleConsumerFactory simpleConsumerFactory,
-      SimpleConsumerManager simpleConsumerManager,
-      KafkaConsumerManager kafkaConsumerManager
-  ) {
-    config.register(new ZkExceptionMapper(appConfig));
+  protected void setupInjectedResources(Configurable<?> config, KafkaRestConfig appConfig,
+                                        ZkUtils zkUtils, MetadataObserver mdObserver,
+                                        ProducerPool producerPool,
+                                        ConsumerManager consumerManager,
+                                        SimpleConsumerFactory simpleConsumerFactory,
+                                        SimpleConsumerManager simpleConsumerManager,
+                                        KafkaConsumerManager kafkaConsumerManager) {
 
-    if (zkUtils == null) {
-      zkUtils = ZkUtils.apply(
-          appConfig.getString(KafkaRestConfig.ZOOKEEPER_CONNECT_CONFIG),
-          30000,
-          30000,
-          JaasUtils.isZkSecurityEnabled()
-      );
-    }
-    if (mdObserver == null) {
-      mdObserver = new MetadataObserver(appConfig, zkUtils);
-    }
-    if (producerPool == null) {
-      producerPool = new ProducerPool(appConfig, zkUtils);
-    }
-    if (consumerManager == null) {
-      consumerManager = new ConsumerManager(appConfig, mdObserver);
-    }
-    if (simpleConsumerFactory == null) {
-      simpleConsumerFactory = new SimpleConsumerFactory(appConfig);
-    }
-    if (simpleConsumerManager == null) {
-      simpleConsumerManager =
-          new SimpleConsumerManager(appConfig, mdObserver, simpleConsumerFactory);
-    }
-    if (kafkaConsumerManager == null) {
-      kafkaConsumerManager = new KafkaConsumerManager(appConfig);
-    }
+    config.register(new ZkExceptionMapper(appConfig));
 
-    this.zkUtils = zkUtils;
-    context =
-        new Context(
-            appConfig,
-            mdObserver,
-            producerPool,
-            consumerManager,
-            simpleConsumerManager,
-            kafkaConsumerManager
-        );
+    DefaultContextProvider.initializeDefaultContext(zkUtils, appConfig, mdObserver, producerPool,
```

Comment:
is it necessary to make the default context here even when the `rest-proxy-context` property is set?